third_party/move/move-binary-format/src/constant.rs

```diff
@@ -38,6 +38,7 @@ fn ty_to_sig(ty: &MoveTypeLayout) -> Option<SignatureToken> {
         MoveTypeLayout::Vector(v) => Some(SignatureToken::Vector(Box::new(ty_to_sig(v.as_ref())?))),
         MoveTypeLayout::Struct(_) => None,
         MoveTypeLayout::Bool => Some(SignatureToken::Bool),
+        MoveTypeLayout::Tagged(_, _) => None,
```

Comment:
What is this thing?

Reply:
This is for constants: when you create a new constant you also have its type in the file / bytecode, and since constants are primitive types, structs (as above) have None and for our use cases aggregators / snapshots are also None. Let's still recursively call `ty_to_sig` just in case + add a comment